nitro-protocol/test/contracts/libraries/Consensus/Consensus.test.ts

```diff
@@ -0,0 +1,103 @@
+import {it} from '@jest/globals';
+import {Contract, Wallet} from 'ethers';
+import {expectRevert} from '@statechannels/devtools';
+
+import testConsensusArtifact from '../../../../artifacts/contracts/test/TESTConsensus.sol/TESTConsensus.json';
+import {
+  generateParticipants,
+  getCountingAppContractAddress,
+  getRandomNonce,
+  getTestProvider,
+  setupContract,
+} from '../../../test-helpers';
+import {TESTConsensus} from '../../../../typechain-types';
+import {
+  Channel,
+  getFixedPart,
+  Outcome,
+  shortenedToRecoveredVariableParts,
+  State,
+  TurnNumToShortenedVariablePart,
+} from '../../../../src';
+import {
+  NOT_UNANIMUS,
+  PROOF_SUPPLIED,
+} from '../../../../src/contract/transaction-creators/revert-reasons';
+import {separateProofAndCandidate} from '../../../../src/contract/state';
+import {expectSucceed} from '../../../expect-succeed';
+const provider = getTestProvider();
+let Consensus: Contract & TESTConsensus;
+
+const chainId = process.env.CHAIN_NETWORK_ID;
+const challengeDuration = 0x1000;
+const asset = Wallet.createRandom().address;
+const defaultOutcome: Outcome = [{asset, allocations: [], metadata: '0x'}];
+const appDefinition = getCountingAppContractAddress();
```

Comment:
Do we want to use the counting app when checking the consensus app?

Reply:
The address does not play any role here, except for creating the `FixedPart`. It seems more logical to use consensus app address here though.

Reply:
36912f830aed69c22bd8f7c23c9c0405eb88f2b9